src/core.js

```diff
@@ -18,6 +33,15 @@ jQuery.fn.init = function( arg1 ) {
 		args[ 0 ] = [];
 	}
 
+	if ( typeof arg1 === "string" && arg1.trim().charAt( 0 ) === "<" ) {
+		if ( !rsingleTag.test( arg1 ) ) {
+			changed = arg1.replace( rxhtmlTag, "<$1></$2>" );
+			if ( changed !== arg1 && makeMarkup( arg1 ) !== makeMarkup( changed ) ) {
+				migrateWarn( "HTML tags must be properly nested and closed: " + arg1 );
```

Comment:
I was thinking about doing it by overwriting `jQuery.htmlPrefilter` instead; then we would miss no cases & the logic would be simpler. Of course, when someone overwrites it on their end, that would remove the warnings but maybe that's acceptable? We could make it more explicit in the [3.5 upgrade guide](https://jquery.com/upgrade-guide/3.5/) that if Migrate is used, it's best to avoid manually overwriting the method?

Doing it in `jQuery.fn.init` probably catches the majority of usage but I suspect it leaves a significant portion out. Consider e.g.:
```js
jQuery( "#test" ).append( "<div/><span/>" );
```
We'd get no warning for that.